lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -44,18 +44,21 @@ local function playerIsInBuild( ply )
     return !getPlayerPvpMode( ply )
 end
 
-local function movementMultiplier(weaponCount) 
+local function movementMultiplier(weaponCount)
+    if weaponCount < 1 then return 1 end
     return math.Clamp(1 - (1.9^( weaponCount))/100, 0, 1)
 end
 
 local function setSpeed(ply, multiplier) 
     -- set speed multiplier, 0 - 1
-    ply:SetRunSpeed(math.Clamp(baseRunSpeed*multiplier,minRunSpeed, baseRunSpeed))
-    ply:SetWalkSpeed(math.Clamp(baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed))
+    local runSpeed = baseRunSpeed * multiplier
+    local walkSpeed = baseWalkSpeed * multiplier
+    ply:SetRunSpeed( math.Clamp( runSpeed, minRunSpeed, baseRunSpeed ) )
+    ply:SetWalkSpeed( math.Clamp( walkSpeed, minWalkSpeed, baseWalkSpeed ) )
 end
 
```

Comment:
I'd like to see a better method name. `adjustMovementSpeed` and `setSpeed` seem similar to me and I'm not 100% sure what each one should do

Reply:
Could just get rid of setSpeed and just put all the code in adjustMovementSpeed. or maybe a name like setRunWalkSpeed?